.genai/io/capstone/genai/ai_bot.py

```
@@ -1,68 +1,217 @@
+        """Get the system message for AI reviewer."""
+        return AiBot.__system_message
+
+    @staticmethod
+    def build_ask_text(code: str, diffs: str) -> str:
+        """
+        Build prompt WITHOUT RAG context (backwards compatible).
+
+        Args:
+            code: Full file content (currently unused but kept for API compatibility)
+            diffs: Git diff output
+
+        Returns:
+            Formatted prompt string
+        """
+        return AiBot.__prompt_simple.format(
+            no_response=AiBot.__no_response,
+            diffs=diffs,
+        )
+
     @staticmethod
-    def build_ask_text(code, diffs) -> str:
-        return AiBot.__chat_gpt_ask_long.format(
-            problems=AiBot.__problems,
+    def build_ask_text_with_context(
+        code: str,
+        diffs: str,
+        rag_context: str,
+        file_path: str = ""
+    ) -> str:
+        """
+        Build prompt WITH RAG context.
+
+        Args:
+            code: Full file content
+            diffs: Git diff output
+            rag_context: Formatted context from RAG retrieval
+            file_path: Path to file being reviewed
+
+        Returns:
+            Formatted prompt string with context
+        """
+        return AiBot.__prompt_with_context.format(
             no_response=AiBot.__no_response,
+            code=code,
             diffs=diffs,
-            # code=code,
+            rag_context=rag_context if rag_context else "(No additional context available)",
+            file_path=file_path if file_path else "(unknown)",
         )
 
     @staticmethod
     def is_no_issues_text(source: str) -> bool:
-        target = AiBot.__no_response.replace(" ", "")
-        source_no_spaces = source.replace(" ", "")
-        return source_no_spaces.startswith(target)
+        """Check if response indicates no issues found."""
+        target = AiBot.__no_response.replace(" ", "").lower()
+        source_no_spaces = source.replace(" ", "").lower()
```

175 : The default implementation of `ai_request_with_context` silently discards the `rag_context` and `file_path` arguments by calling `ai_request_diffs`. This can cause silent failures where a caller expects a context-aware review but receives a basic one if the bot subclass has not overridden the method. To make the behavior more explicit and robust, consider making `ai_request_with_context` the primary abstract method that subclasses must implement.